packages/ptokens-swap/src/ptokens-swap-builder.ts

```diff
@@ -126,12 +182,30 @@ export class pTokensSwapBuilder {
    * Build a pTokensSwap object from the parameters set when interacting with the builder.
    * @returns - An immutable pTokensSwap object.
    */
-  build(): pTokensSwap {
+  async build(): Promise<pTokensSwap> {
     if (!this._sourceAsset) throw new Error('Missing source asset')
     if (this._destinationAssets.length === 0) throw new Error('Missing destination assets')
     if (!this._amount) throw new Error('Missing amount')
     if (!this.isValidSwap()) throw new Error('Invalid swap')
-    const ret = new pTokensSwap(this.sourceAsset, this._destinationAssets, this._amount)
-    return ret
+    if (!this._interimNetworkId) this.setInterimChain(INTERIM_CHAIN_NETWORK_ID)
+    if (!this._interimProvider)
+      this.setInterimProvider(
+        new pTokensEvmProvider(
```

Comment:
I would create a helper function in ptokens-assets-evm/lib/index,
something like `getDefaultProvider(_networkId)`

Reply:
In this way, viem is restricted to ptokens-assets-evm